stellargraph/core/graph.py

```diff
@@ -722,7 +724,7 @@ def check_graph_for_ml(self, features=True):
 
         # TODO: check the feature node_ids against the graph node ids?
 
-    def node_features(self, nodes, node_type=None):
+    def node_features(self, nodes, node_type=None) -> np.ndarray:
```

Comment:
This is a large amount of duplication of subtle code!

Maybe this one could be entirely replaced with something like:

```python
return self.node_features_tensor(nodes, node_type).numpy()
```

Reply:
unfortunately that won't work for non eager mode - but if we decide to not support it then yeah definitely!

Reply:
We could wrap it in the `Session()` magic that you're doing inside this function, I think?

Reply:
good idea!